packages/matter.js/src/cluster/server/CommandServer.ts

```diff
@@ -49,14 +54,39 @@ export class CommandServer<RequestT, ResponseT> {
         /** Response data */
         response: TlvStream;
     }> {
-        const request = this.requestSchema.decodeTlv(args);
-        logger.debug(`Invoke ${this.name} with data ${Logger.toJSON(request)}`);
-        const response = await this.handler(request, session, message, endpoint);
-        logger.debug(`Invoke ${this.name} response : ${Logger.toJSON(response)}`);
-        return {
-            code: StatusCode.Success,
-            responseId: this.responseId,
-            response: this.responseSchema.encodeTlv(response),
-        };
+        return await tryCatchAsync(
+            async () => {
+                let request = this.requestSchema.decodeTlv(args);
+
+                // Inject fabric index into structures in general if undefined, if set it will be used
+                if (session.isSecure()) {
+                    const fabric = (session as SecureSession<any>).getFabric();
+                    if (fabric) {
+                        request = this.requestSchema.injectField(
+                            request,
+                            <number>Globals.FabricIndex.id,
+                            session.getAssociatedFabric().fabricIndex,
+                            existingFieldIndex => existingFieldIndex === undefined,
+                        );
+                    }
+                }
+
+                this.requestSchema.validate(request);
+                logger.debug(`Invoke ${this.name} with data ${Logger.toJSON(request)}`);
+                const response = await this.handler(request, session, message, endpoint);
+                logger.debug(`Invoke ${this.name} response : ${Logger.toJSON(response)}`);
+                return {
+                    code: StatusCode.Success,
+                    responseId: this.responseId,
+                    response: this.responseSchema.encodeTlv(response),
+                };
+            },
+            ValidationError,
+            async (error: ValidationError) => {
+                const statusError = new StatusResponseError(error.message, StatusCode.ConstraintError);
```

Comment:
Could we make ValidationError a StatusResponseError? I've been doing that with pretty much any error that could be triggered by remote interaction.

If it's thrown locally doesn't seem like it'd be a problem. But if it doesn't seem appropriate to throw "status response" locally we could maybe move the status code into the base error class or rename StatusResponseError.

Reply:
We have some places where we check if an error is StatusResponseError because we know itthat only can be an internal error ... we would need to adjust these places then to differentiate "generated" StatusResponseErrors from "received" ones ...

I could do that .... WDYT?

Reply:
Right so if we have a ValidationError on input it should go to the wire but on output we should handle more as an internal error.  Is that what you're saying?

Reply:
Ok I checked all places. Adjusted ... PR gets bigger now because I needed to resolve circular imports. Please have a look